### PR TITLE
[yang] add NEIGH yang model

### PR DESCRIPTION
**Pull request by Ndancejic**
#### Why I did it
Yang model for NEIGH table was missing
Fixed https://github.com/sonic-net/sonic-buildimage/issues/13971

#### How I did it
added sonic-neigh.yang model

#### How to verify it
make buildimage

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Description for the changelog
Adding NEIGH yang model

#### Link to config_db schema for YANG module changes

(https://github.com/sonic-net/sonic-buildimage/blob/b721ff87b976a6a38bdd65443ea3bc686014e783/src/sonic-yang-models/doc/Configuration.md#neigh)

### CODE REVIEW

**Review by ganglyu — APPROVED** [review]

**Comment by zhangyanzhao** [comment]
Thank you @Ndancejic  for defining this YANG model to unblock the stop-bleeding feature enablement.

**Comment by isabelmsft** [comment]
I suspect generic_config_updater unit test failures are related to libyang backlink issue: https://github.com/sonic-net/sonic-buildimage/issues/9312

**Comment by wen587** [comment]
Hi @Ndancejic   you may need to comment out vlan import. As it will break libyang backlinks. Issue can be tracked here: https://github.com/sonic-net/sonic-buildimage/issues/9312

**Comment by zhangyanzhao** [comment]
@Ndancejic  can you please check this issue and update? This is blocking YANG model work. Thanks.

File: src/sonic-yang-models/yang-models/sonic-neigh.yang [hunk 1]
```diff
@@ -31,14 +27,14 @@ module sonic-neigh {
 				key "vlan neighbor";
 
 				leaf vlan {
-					description "Neighbor interface and IP address ex. Vlan1000|100.1.1.3";
-					type leafref {
-						path "/svlan:sonic-vlan/svlan:VLAN/svlan:VLAN_LIST/svlan:name";
+					description "Neighbor Vlan interface ex. Vlan1000";
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-neigh.yang:30]: https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-vxlan.yang#L87-L90
Would please add comment like above example?
When we fix back-links issue, we need to remove this workaround.

**Review by ganglyu — APPROVED** [review]

**Pull request merged**